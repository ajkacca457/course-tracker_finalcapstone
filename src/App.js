import '@fortawesome/fontawesome-free/css/all.min.css';
import {
  BrowserRouter as Router, Route, Switch, Redirect,
} from 'react-router-dom';
import Home from './components/home';
import Header from './components/header';
import Login from './components/login';
import Signup from './components/signup';
import Addcourse from './components/addcourse';
import Coursecontainer from './containers/coursecontainer';
import Singlecourse from './components/singlecourse';
import Progress from './containers/reportcont';
import Protected from './components/protected';

const App = () => (
  <Router>
    <div className="App">
      <Header />
      <Switch>
        <Route exact path="/">
          <Redirect to="/login" />
        </Route>
        <Route exac path="/login" component={Login} />
        <Route exac path="/signup" component={Signup} />
        <Route exac path="/home">
          <Protected Cmp={Home} />
        </Route>
        <Route exac path="/addcourse">
          <Protected Cmp={Addcourse} />
        </Route>

        <Route exac path="/courses/:id">
          <Protected Cmp={Singlecourse} />
        </Route>

        <Route exac path="/courses">
          <Protected Cmp={Coursecontainer} />
        </Route>

        <Route exac path="/progress">
          <Protected Cmp={Progress} />
        </Route>

      </Switch>
    </div>
  </Router>
);

export default App;
