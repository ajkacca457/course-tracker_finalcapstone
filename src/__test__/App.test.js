import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';
import App from '../App';

describe('App Component', () => {
  it('renders correctly', () => {
    const apptree = TestRenderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(apptree).toMatchSnapshot();
  });
});
