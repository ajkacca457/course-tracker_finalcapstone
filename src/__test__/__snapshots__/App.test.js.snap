// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="header bg-dark px-2 my-0 d-flex justify-content-between align-items-center"
  >
    <h4
      className="text-center text-white pt-1 mb-3"
    >
      <i
        className="fas fa-chalkboard-teacher"
      />
       
      Course Tracker
    </h4>
    <button
      className="d-none"
      onClick={[Function]}
      type="button"
    >
       
      <i
        className="fas fa-power-off text-white"
      />
    </button>
  </div>
  <div
    className="logincont"
  >
    <div
      className="bg-white mx-auto p-2 rounded loginitem"
    >
      <h5
        className="text-center bg-info py-2 text-white mb-3 rounded"
      >
        Login with email and password
      </h5>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group text-left"
        >
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Enter username"
            type="text"
          />
        </div>
        <div
          className="form-group text-left"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
        <button
          className="btn btn-info w-100"
          type="submit"
        >
          Login
        </button>
      </form>
      <div
        className="signupinfo text-right"
      >
        <p
          className="tagline"
        >
          Dont have an account?
        </p>
        <a
          className="nav-link m-0 bg-danger w-25 ml-auto text-center text-white tagbtn"
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </div>
    </div>
  </div>
</div>
`;
